### PR TITLE
Fixes #3321 - Converts to pytest and improves coverage

### LINKED ISSUE #3321: [test] webcompat/issues.py needs more tests coverage
coverage is missing currently 

```
Name                                   Stmts   Miss  Cover   Missing
--------------------------------------------------------------------
webcompat/issues.py                       48      4    92%   99-104
```

https://github.com/webcompat/webcompat.com/blob/54a27f0a1b0968195e31d3c0c447593d5011af18/webcompat/issues.py#L99-L104

Probably two steps.

* [x] convert to pytest (see [this diff as an example](https://github.com/webcompat/webcompat.com/pull/3312/commits/d6c87dfee3cd23b004b137e1fbd4df5010ebca35) and [documentation](https://github.com/webcompat/webcompat.com/blob/master/docs/tests.md#python-unit-tests)
* [x] add the remaining missing tests scenario

the file to edit is 
https://github.com/webcompat/webcompat.com/blob/master/tests/unit/test_issues.py

### PR DESCRIPTION
**Pull request by karlcow**
This PR fixes issue #3321 

## Proposed PR background

Please provide enough information so that others can review your pull request:

This took a bit more time than I was expecting but I have figured out a couple of things. 
This also should lead to more test organization. I need to open an issue so that we put the mock_api_response outside of test_api_urls.  (probably in conftest.py if I understood well pytest) 

Anyway.

It basically 
1. Converts the test for test_issues.py to pytest
2. Adds missing tests for cases which were not covered in the tests (thanks to coverage for finding them.)
3. Modifies mock_api_response to add `.json()` method for the mocks.

Now we have 100% coverage for the issues.py module. See #3205 for the work to do still.
 
```
(env) ~/code/webcompat.com % COLUMNS=50 coverage report -m                                            
Name                                   Stmts   Miss  Cover   Missing
--------------------------------------------------------------------
…
webcompat/issues.py                       48      0   100%
…
--------------------------------------------------------------------
TOTAL                                   1897    343    82%
```

ok let's see if we stay green or if we broke something.

### CODE REVIEW

File: tests/unit/test_issues.py [hunk 1]
```diff
@@ -6,109 +6,145 @@
...
+FORM_AUTH = FORM.copy()
+FORM_AUTH['submit_type'] = 'github-auth-report'
+
+
+@pytest.fixture
+def client():
+    """Test client for requests."""
+    webcompat.app.config['TESTING'] = True
+    with webcompat.app.test_client() as client:
+        yield
```
Comment (miketaylr) [inline tests/unit/test_issues.py:46]: [In this example](https://flask.palletsprojects.com/en/1.1.x/testing/#the-testing-skeleton), it `yield`s the `client`. Is there a specific reason we don't do that here?

I see none of these tests actually use the `test_client`, so I guess all this fixture is doing is setting testmode. Probably better to yield the `client` I think.

File: tests/unit/test_issues.py [hunk 2]
```diff
@@ -6,109 +6,145 @@
...
+
+
+def test_report_with_auth_on(client):
+    """Test that we can still work with github-auth-report."""
+    with patch('webcompat.issues.github.post') as json_resp:
+        json_resp.return_value = {'number': 2}
+        json_data = report_issue(FORM_AUTH, False)
```
Comment (miketaylr) [inline tests/unit/test_issues.py:53]: Can we pass in the second arg as a kwarg? I think `report_issue(FORM_AUTH, proxy=False)` is easier to understand than `report_issue(FORM_AUTH, False)`:

https://github.com/webcompat/webcompat.com/blob/7ada3d5f34b848d88d60843dc8b824533c357df8/webcompat/issues.py#L88

File: tests/unit/test_issues.py [hunk 3]
```diff
@@ -6,109 +6,145 @@
...
+    """Test issue posting report the right json."""
+    with patch('webcompat.issues.proxy_request') as github_data:
+        github_data.return_value = mock_api_response({
+            'status_code': 201,
+            'content': '[{"number":2}]',
+        })
+        with patch('webcompat.issues.report_private_issue') as silent:
+            # We avoid making a call to report_private_issue
+            silent.return_value = None
+            json_data = report_issue(FORM_ANON, True)
```
Comment (miketaylr) [inline tests/unit/test_issues.py:68]: (same comment here for `report_issue`)

File: tests/unit/test_issues.py [hunk 4]
```diff
@@ -6,109 +6,145 @@
...
+            assert type(json_data) is dict
+            assert json_data.get('number') == 2
+
+
+def test_report_issue_fails_400_for_unknown_type(client):
+    """Test request fails with a 400 because type unknown."""
+    with pytest.raises(BadRequest):
+        rv = report_issue(FORM, True)
```
Comment (miketaylr) [inline tests/unit/test_issues.py:76]: (same comment here for `report_issue`)

File: tests/unit/test_issues.py [hunk 5]
```diff
@@ -6,109 +6,145 @@
...
+
+def test_report_issue_fails_400_for_proxy(client):
+    """Test github-proxy-report fails with a 400."""
+    with patch('webcompat.issues.proxy_request') as github_data:
+        github_data.return_value = mock_api_response({
+            'status_code': 400,
+            'content': '[{"number":2}]',
+        })
+        with pytest.raises(BadRequest):
+            rv = report_issue(FORM_ANON, True)
```
Comment (miketaylr) [inline tests/unit/test_issues.py:87]: (same comment here for `report_issue`)

File: tests/unit/test_issues.py (hunk 1 above)
Comment (karlcow) [reply tests/unit/test_issues.py:46]: Good point I will remove it entirely.

File: tests/unit/test_issues.py (hunk 2 above)
Comment (karlcow) [reply tests/unit/test_issues.py:53]: will do for the four of them. Better being explicit indeed.

File: tests/unit/test_issues.py (hunk 3 above)
Comment (karlcow) [reply tests/unit/test_issues.py:68]: see https://github.com/webcompat/webcompat.com/pull/3322#discussion_r433003788

File: tests/unit/test_issues.py (hunk 4 above)
Comment (karlcow) [reply tests/unit/test_issues.py:76]: see https://github.com/webcompat/webcompat.com/pull/3322#discussion_r433003788

File: tests/unit/test_issues.py (hunk 5 above)
Comment (karlcow) [reply tests/unit/test_issues.py:87]: see https://github.com/webcompat/webcompat.com/pull/3322#discussion_r433003788

**Comment by karlcow** [comment]
ok rebased. and pushed.

File: tests/unit/test_issues.py [hunk 6]
```diff
@@ -6,109 +6,143 @@
...
+    ('username', ''), ])
+
+FORM_ANON = FORM.copy()
+FORM_ANON['submit_type'] = 'github-proxy-report'
+FORM_AUTH = FORM.copy()
+FORM_AUTH['submit_type'] = 'github-auth-report'
+
+
+@pytest.fixture
+def client():
```
Comment (miketaylr) [inline tests/unit/test_issues.py:42]: nit: maybe better to rename this method to `setup` or something now?

Comment (karlcow) [reply tests/unit/test_issues.py]: good idea. done.

**Comment by miketaylr** [comment]
Thanks Karl, very nice improvements.

**Pull request merged**